src/Swift.Runtime/src/Swift/SwiftOptional.cs

```diff
@@ -99,7 +101,8 @@ public static SwiftOptional<T> NewSome(T value)
             fixed (byte* payload = instance._payload)
             {
                 var metadata = SwiftObjectHelper<SwiftOptional<T>>.GetTypeMetadata();
-                SwiftMarshal.MarshalToSwift(value, new IntPtr(payload));
+                Span<byte> payloadSpan = new Span<byte>(payload, (int)metadata.Size - 1);
```

Comment:
Why `- 1`?

Reply:
The payload of `Optional` is the size of the type minus 1 for the optional tag.

Reply:
Thanks. Just so I'm clear is this one byte? one pointer? What unit is the "1" in this case? If this is an ABI contract can you add a reference for it?

Reply:
Yes, it should be one byte. Added a reference.